docgen/src/guide/React_native.md

```diff
@@ -6,9 +6,643 @@ category: guide
+
+Since the dataset used here is an e-commerce one, let's add a RefinementList using the [connectRefinementList](connectors/connectRefinementList.html) connector
+to filter the products by categories.
+
+This connector gives you two interesting properties: 
+
+* `items `, the list of refinements. Each item has a `isRefined` property indicating if they are selected or not. 
+* `refine`, the function to call when a new category is selected
+
+On the React Native side we are going to take advantage of the [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) and the [TouchableHighlight](http://facebook.github.io/react-native/releases/0.45/docs/touchablehighlight.html#touchablehighlight) component to render this refinement list. 
+
+```jsx
+// We need to add the RefinementList to our import
+import {
+  connectInfiniteHits,
+  connectSearchBox,
+  connectHighlight,
+  connectRefinementList,
+} from 'react-instantsearch/connectors';
+//We need to add the TouchableHighlight to our import
+import {
+  StyleSheet,
+  View,
+  FlatList,
+  Image,
+  Text,
+  TextInput,
+  TouchableHighlight,
+} from 'react-native';
+
+// [...]
+
+const RefinementList = connectRefinementList(({ refine, items }) => {
+  const keyExtractor = (item, index) => index;
+
+  return (
+    <FlatList
+      data={items}
+      keyExtractor={keyExtractor}
+      ListHeaderComponent={() =>
+        <Text style={{ marginTop: 20, height: 50, alignSelf: 'center' }}>
+          Categories
+        </Text>}
+      renderItem={({ item }) => {
+        return (
+          <View style={{ height: 30 }}>
+            <TouchableHighlight
+              onPress={() => {
+                refine(item.value);
+              }}
+            >
+              <Text style={item.isRefined ? { fontWeight: 'bold' } : {}}>
+                {item.label}
+              </Text>
+            </TouchableHighlight>
+          </View>
+        );
+      }}
+    />
+  );
+});
+```
+
+Let’s add this new `<RefinementList>` component to our InstantSearch instance. 
+
+```jsx
+const App = () =>
+   <View style={styles.container}>
+        <InstantSearch
+            appId="latency"
+            apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+            indexName="ikea"
+         >
+            <View
+            style={{
+                flexDirection: 'row',
+            }}
+            >
+            <SearchBox />
+            </View>
+            <RefinementList attributeName="category" />
+            <Hits />
+        </InstantSearch>
+    </View>
+```
+
+The `attributeName` props specifies the faceted attribute to use in this widget. This attribute should be declared as a facet in the index configuration as well.
+
+The values displayed are computed by Algolia from the results.
+
+So far, we've seen the following:
+- there are components for all types of refinements
+- the RefinementList works with facets
+- facets are computed from the results
+
+## Putting our Refinement List in a Modal
```

Comment:
first time you spell RefinementList in two words, probably better as `RefinementList`